examples/fluids/navierstokes.c

```diff
@@ -280,26 +281,50 @@ int main(int argc, char **argv) {
     const CeedInt  num_P = app_ctx->degree + 1, num_Q = num_P + app_ctx->q_extra;
     // -- Get global size
-    PetscCall(VecGetSize(Q, &glob_dofs));
-    PetscCall(VecGetLocalSize(Q, &owned_dofs));
-    glob_nodes = glob_dofs / num_comp_q;
+    PetscCall(DMGetGlobalVectorInfo(user->dm, &owned_dofs, &glob_dofs, NULL));
     // -- Get local size
-    PetscCall(VecGetSize(user->Q_loc, &local_nodes));
-    local_nodes /= num_comp_q;
+    PetscCall(DMGetLocalVectorInfo(user->dm, &local_dofs, NULL, NULL));
     PetscCall(PetscPrintf(comm,
                           "  Mesh:\n"
                           "    Number of 1D Basis Nodes (P)       : %" CeedInt_FMT "\n"
                           "    Number of 1D Quadrature Points (Q) : %" CeedInt_FMT "\n"
                           "    Global DoFs                        : %" PetscInt_FMT "\n"
-                          "    Owned DoFs                         : %" PetscInt_FMT "\n"
                           "    DoFs per node                      : %" PetscInt_FMT "\n"
-                          "    Global nodes (DoFs / %" PetscInt_FMT ")            : %" PetscInt_FMT "\n"
-                          "    Local nodes                        : %" PetscInt_FMT "\n",
-                          num_P, num_Q, glob_dofs, owned_dofs, num_comp_q, num_comp_q, glob_nodes, local_nodes));
+                          "    Global nodes                       : %" PetscInt_FMT "\n",
+                          num_P, num_Q, glob_dofs, num_comp_q, glob_dofs / num_comp_q));
+    // -- Get Partition Statistics
+    PetscInt sendbuf[2], part_owned_dofs[2], part_local_dofs[2];
+    sendbuf[0] = sendbuf[1] = owned_dofs;
+    PetscCall(PetscGlobalMinMaxInt(comm, sendbuf, part_owned_dofs));
+    sendbuf[0] = sendbuf[1] = local_dofs;
+    PetscCall(PetscGlobalMinMaxInt(comm, sendbuf, part_local_dofs));
+    PetscReal part_owned_dof_ratio = (PetscReal)part_owned_dofs[1] / (PetscReal)part_owned_dofs[0];
+    PetscReal part_local_dof_ratio = (PetscReal)part_local_dofs[1] / (PetscReal)part_local_dofs[0];
+    PetscCall(PetscPrintf(comm,
+                          "  Partition:\n"
+                          "    Owned free nodes (min,max,ratio)   : %" PetscInt_FMT ", %" PetscInt_FMT ", %f\n"
+                          "    Local nodes (min,max,ratio)        : %" PetscInt_FMT ", %" PetscInt_FMT ", %f\n"
+                          "(nodes == DoFs / %" PetscInt_FMT ")\n",
+                          part_owned_dofs[0] / num_comp_q, part_owned_dofs[1] / num_comp_q, part_owned_dof_ratio, part_local_dofs[0] / num_comp_q,
+                          part_local_dofs[1] / num_comp_q, part_local_dof_ratio, num_comp_q));
+    {
+      PetscSF            sf;
+      PetscInt           niranks;
+      const PetscInt    *ioffset, *irootloc;
+      const PetscMPIInt *iranks;
+      PetscCall(DMGetSectionSF(user->dm, &sf));
+      PetscCall(PetscSFGetLeafRanks(sf, &niranks, &iranks, &ioffset, &irootloc));
```

Comment:
I was going for "what do I have that is owned by someone else", which I think is the same as "who owns stuff that I need".

I'm not sure if we want both though. I assume that the primary latency for MPI communication is "waiting to recieve" rather than "waiting to send", but I suppose if there's a significant concentration of sends in on a specific network switch, that could cause bottlenecks.